### PR TITLE
Feat bookmarks part1

### PR DESCRIPTION
**Pull request by dangeol**
This is the bookmarks feature, without the bookmarks list page, which shall be implemented in part 2. I won't be available within the upcoming two weeks.

### CODE REVIEW

File: tools/migration/funcInfo0.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
...
..."DPRShowBottomPane","getSelected","convert","changeConversion","clearboth","sendtoconvert","sendtoPad","clearPad","savePad","moveframex","moveframec","closeCP","openCP","cpFlatten","showHideId","moves","moveanf","scrolldown","getlink","go_anchor","scrollToId","bookmarkXML","eraseBookmark","eraseBookmarks","bookmarkframe","bookmarkxd","bookmarkxn","convertOldBookmarks","saveBookmark","makeDeclensions","outputAnalysis","analyzeword","findmatch","manualCompound","isIndec","isUncomp","outputDef","showShortDef","conciseChange","reanalyze","conjugate","conjugateNoun","conjugateVerb","conjugateIrrNoun","conjugateIrrVerb","conjugateWord","formatConjugation","getConjugation","makeGrammarTerms","typeAndGender","conjugateWords","bv","showBv","citation","bvAlert","translateTextx","translateText","arrangeWords","checkCompatibleNoun","checkCompatibleVerb","makeWord","translateWord","transMod","addPlural","addPhrasePreps","sortLongerDec","stripEnglish","simpleWordTranslation","showAltTable","changeAlt","insertWordByWord","clearText","insertConj","DPR_keypress","createCommands","moveFrame","devO","dalert","ddump","mainInitialize","installGlobalHandlers","triggerUpdateCheck","setupBTForRG","dump"],
```
Comment (parthopdas) [inline tools/migration/funcInfo0.json:2]: any idea why this file changed?

File: _dprhtml/features/other-dialogs/init.js [hunk 2]
```diff
@@ -93,12 +93,54 @@ class OtherDialogsViewModel {
...
-    this.bookmarkInput('');
     $('#bookmark-dialog-root').modal('show');
   }
 
-  sendBookmarkFromDialog(loc, name, desc) {
-    // TODO when bookmarks is implemented
+  sendBookmarkFromDialog() {
+
+    var loc = __navigationTabViewModel.placeArray();
+    var name = $('#dialog-bookmarkInput').val();
```
Comment (parthopdas) [inline _dprhtml/features/other-dialogs/init.js:102]: there is ko binding already on dialog-bookmarkInput - lets use that and not jquery

File: _dprhtml/js/bookmarks.js [hunk 3]
```diff
@@ -1,39 +1,57 @@
...
-  {
-        eraseFile('DPR_Bookmarks');
-    var sidebar = DPRSidebarWindow();
-    if (sidebar) {
-      sidebar.DPRNav.bookmarkBox();
+function addBookmark() {
+  if (__navigationTabViewModel.isStorageSupportedByBrowser) {
+    let bookmarksArrayFromStorage = localStorage.getItem("bookmarksArray");
+    if (bookmarksArrayFromStorage) {
+      let value = `${$('#dialog-bookmarkInput').val()}@${__navigationTabViewModel.locPlace}`;
```
Comment (parthopdas) [inline _dprhtml/js/bookmarks.js:46]: there is ko binding already on dialog-bookmarkInput - lets use that and not jquery

in general - lets try to get away from hard coding IDs across the files - it makes the product brittle

File: _dprhtml/js/format.js [hunk 4]
```diff
@@ -458,6 +458,7 @@ function convtitle(nikaya,book,una,vna,wna,xna,yna,zna,hiert,oneline,click)
     }
 
     var onet = '<b style="color:'+DPR_G.DPR_prefs[col[w++]]+'">' + translit(toUni(thisname)) + '</b>';
+    $('#dialog-bookmarkInput').val(translit(toUni(thisname)));
```
Comment (parthopdas) [inline _dprhtml/js/format.js:461]: there is ko binding already on dialog-bookmarkInput - lets use that and not jquery

File: _dprhtml/features/navigation/init.js [hunk 5]
```diff
@@ -59,14 +59,21 @@ class NavigationTabViewModel {
     this.para = ko.observable('');
 
     this.places = ko.observableArray();
+    this.locPlace = ko.observableArray();
```
Comment (parthopdas) [inline _dprhtml/features/navigation/init.js:62]: can we please give it a more descriptive name?

File: _dprhtml/features/navigation/init.js [hunk 6]
```diff
@@ -91,7 +98,7 @@ class NavigationTabViewModel {
 
   setPlaces(places) {
     this.places(places);
-    this.place(places[0].place)
+    this.place(places[0].place);
```
Comment (parthopdas) [inline _dprhtml/features/navigation/init.js:101]: please revert this change

Comment (dangeol) [reply _dprhtml/features/navigation/init.js:101]: reverted

File: _dprhtml/features/navigation/init.js (hunk 5 above)
Comment (dangeol) [reply _dprhtml/features/navigation/init.js:62]: sectionPlace

File: _dprhtml/js/format.js (hunk 4 above)
Comment (dangeol) [reply _dprhtml/js/format.js:461]: done

File: _dprhtml/js/bookmarks.js (hunk 3 above)
Comment (dangeol) [reply _dprhtml/js/bookmarks.js:46]: done

File: _dprhtml/features/other-dialogs/init.js (hunk 2 above)
Comment (dangeol) [reply _dprhtml/features/other-dialogs/init.js:102]: done

File: tools/migration/funcInfo0.json (hunk 1 above)
Comment (dangeol) [reply tools/migration/funcInfo0.json:2]: I had renamed the function ` bookmarkXML()` to `getBookmarks()`. This file has been changed during a "Replace all" operation. Is this an issue?

Comment (parthopdas) [reply tools/migration/funcInfo0.json:2]: no issues - i was just curious. this is used by the tool to migrate our js files to IEFEs in preparation to migration to es6 modules.
